Move more Local/RemoteDOMWindow functions to DOMWindow

charliewolfe opened the pull request:
#### 8b99375ed38cff1aa03e2c6423c242af1b1669ff
<pre>
Move more Local/RemoteDOMWindow functions to DOMWindow
<a href="https://bugs.webkit.org/show_bug.cgi?id=268564">https://bugs.webkit.org/show_bug.cgi?id=268564</a>
<a href="https://rdar.apple.com/122117399">rdar://122117399</a>

Reviewed by Tim Nguyen and Alex Christensen.

* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::opener const):
(WebCore::DOMWindow::top const):
(WebCore::DOMWindow::parent const):
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::opener const): Deleted.
(WebCore::LocalDOMWindow::parent const): Deleted.
(WebCore::LocalDOMWindow::top const): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/RemoteDOMWindow.cpp:
(WebCore::RemoteDOMWindow::top const): Deleted.
(WebCore::RemoteDOMWindow::opener const): Deleted.
(WebCore::RemoteDOMWindow::parent const): Deleted.
* Source/WebCore/page/RemoteDOMWindow.h:

Canonical link: <a href="https://commits.webkit.org/273964@main">https://commits.webkit.org/273964@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3aaa85411790aecea95b6a87e777f1af2a32d99

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37266 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16163 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39557 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39823 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33233 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38570 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18744 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13338 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31710 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37831 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13629 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32791 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11870 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11853 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33414 "layout-tests (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41083 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/31136 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33732 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33848 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37768 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12207 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10008 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35912 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13825 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12558 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4845 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/page/DOMWindow.cpp:
```diff
@@ -139,4 +139,42 @@ WebCoreOpaqueRoot root(DOMWindow* window)
+    if (localFrame && !localFrame->loader().opener())
+        return nullptr;
+
+    return &frame->windowProxy();
```
This is not correct.  Also this function doesn't need a dynamicDowncast<LocalFrame> because Frame::opener exists.

charliewolfe replied:
Whoops, will switch this to the opener window proxy.

charliewolfe committed:
Move more Local/RemoteDOMWindow functions to DOMWindow
https://bugs.webkit.org/show_bug.cgi?id=268564
rdar://122117399

Reviewed by Tim Nguyen and Alex Christensen.

* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::opener const):
(WebCore::DOMWindow::top const):
(WebCore::DOMWindow::parent const):
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::opener const): Deleted.
(WebCore::LocalDOMWindow::parent const): Deleted.
(WebCore::LocalDOMWindow::top const): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/RemoteDOMWindow.cpp:
(WebCore::RemoteDOMWindow::top const): Deleted.
(WebCore::RemoteDOMWindow::opener const): Deleted.
(WebCore::RemoteDOMWindow::parent const): Deleted.
* Source/WebCore/page/RemoteDOMWindow.h:

Canonical link: https://commits.webkit.org/273964@main
